src/core/Akka.Remote/RemoteSystemDaemon.cs

```diff
@@ -258,37 +258,27 @@ private void HandleDaemonMsgCreate(DaemonMsgCreate message)
         /// <returns>ActorRef.</returns>
         public override IActorRef GetChild(IEnumerable<string> name)
         {
-            string[] parts = name.ToArray();
-
-            if (string.IsNullOrEmpty(parts.FirstOrDefault()))
+            var elements = name.ToArray();
+            var path = elements.Join("/");
+            var n = 0;
+            while (true)
             {
-                return this;
-            }
-
-
-            var joined = string.Join("/", parts);
-
-            var child = GetChild(joined);
-            if (child != null)
-            {
-                return child;
-            }
-            var last = joined.LastIndexOf("/", StringComparison.Ordinal);
-            var i = 1;
-            while (last != -1 || parts.Length <= i)
```

Comment:
With a deployed actor path `last` was `-1 ` but `i` was < `parts.Length` so the `Substring` in the next line throws.